services/ui/src/routes/index.svelte

```diff
@@ -1,9 +1,8 @@
 <script>
   import VariableDiag from "$lib/components/VariableDiag";
+  import api from "$lib/helpers/api.helper";
 
-  let data = fetch("http://localhost:5000/diag/temperature").then((response) =>
-    response.json()
-  );
+  let data = api.diagnostics();
```

Comment:
A question - how do `api.helper` interact with `env.helpers` and `src/routes/diag/temperature.js`? 

I'm wondering if this needs to be updated to use the env helpers and new route. When I load the container image in the cluster it looks like it tries to access `localhost/diag/temperature` instead of the host specified in `UG_DIAG_API_HOST`. (e.g. `UG_DIAG_API_HOST="http://dev-graphics-api"`)

I'm debating if I misconfigured the cluster. However, the ui works when I publicly expose the API service on `http://localhost/diag/temperature`

Reply:
The `api.helper.js` doesn’t need to know about the Python host anymore. It makes requests to the JS backend, which is handled by the `diag/temperature.js` route. The `diag/temperature.js` route is already using the `UG_DIAG_API_HOST` environment variable to make its requests. The `env.helper.js` module is just a convenient way to wrap access to environment variables that we use for configuration so we don’t have to sprinkle `process.env` everywhere; it also provides a convenient way to specify default values as needed.

I’m not entirely sure what the issue is you’re describing. If the Python app and the JS app are running in separate containers, then a request to `localhost` on the JS app should resolve to its own container, not the Python container.

Next week I’ll try adding some logging to the JS app so we can see what it thinks it’s supposed to connect to. Maybe that will shed some light.

Reply:
I pushed a change that adds logging to the route. If you set

```
DEBUG="unified-graphics:route"
```

In your environment, you should get some log messages from `/diag/temperature/` which includes the URL it is using to fetch data from the Python application.

Reply:
Awesome. Once I got the root issue figured out, that works!

It turns out that setting the host with `"`'s like `UG_DIAG_API_HOST="http://dev-graphics-api"` is invalid and I needed to use `UG_DIAG_API_HOST=http://dev-graphics-api` instead. Intriguingly, the logging didn't help in this situation - it seems like the logging was never reached and I'd just see the below in the logs.

```console
$ kubectl logs -n dev dev-graphics-ui-786dbb8f9d-8stlm             
Listening on 0.0.0.0:80
```

Reply:
One thing to note might be that the `debug` modules logs to `stderr`, so if k8s is not capturing that, that could be why the logs don’t show up.

Reply:
That's a good point. I checked and the documentation suggests that both `stdout` and `stderr` are captured by k8s logging interface so that shouldn't be the issue. I'll go ahead and split this out into a new issue since I don't think this needs to be held up. It's working in k8s with the changes here as long as you're careful about double quotes.